Source/doomstat.c

```diff
@@ -116,6 +116,28 @@ char *MAPNAME(int e, int m)
   return name;
 }
 
+boolean VANILLAMAP(int e, int m)
```

Comment:
I think this can be broken down into a check for `gamemode == commercial`. It should be safe to e.g. load a E4M1 map with non-ultimate Doom 1.